crates/fluvio-cli/src/client/topic/create.rs

```diff
@@ -203,36 +277,81 @@ mod load {
     use std::path::Path;
 
     use anyhow::{anyhow, Result};
-    use fluvio::metadata::topic::PartitionMaps;
+    use fluvio::metadata::topic::{PartitionMaps, MirrorConfig};
+    use fluvio_controlplane_metadata::topic::TargetMirrorConfig;
 
     pub(crate) trait ReadFromJson: Sized {
         /// Read and decode from json file
-        fn read_from_json_file<T: AsRef<Path>>(path: T) -> Result<Self>;
+        fn read_from_json_file<T: AsRef<Path>>(path: T, topic: &str) -> Result<Self>;
     }
 
     impl ReadFromJson for PartitionMaps {
-        fn read_from_json_file<T: AsRef<Path>>(path: T) -> Result<Self> {
+        fn read_from_json_file<T: AsRef<Path>>(path: T, _topic: &str) -> Result<Self> {
             let file_str: String = read_to_string(path)?;
             serde_json::from_str(&file_str)
                 .map_err(|err| anyhow!("error reading replica assignment: {err}"))
         }
     }
 
+    impl ReadFromJson for MirrorConfig {
+        fn read_from_json_file<T: AsRef<Path>>(path: T, topic: &str) -> Result<Self> {
```

Comment:
add unit test for this

Reply:
I think that I already added at line 329